libqtile/widget/base.py

```diff
@@ -388,13 +388,42 @@ class _TextBox(_Widget):
         ("markup", True, "Whether or not to use pango markup"),
         ("fmt", "{}", "How to format the text"),
         ("max_chars", 0, "Maximum number of characters to display in widget."),
+        (
+            "scroll",
+            False,
+            "Whether text should be scrolled. When True, you must set the widget's ``width``.",
```

Comment:
"When True, you must set the widget's ``width``" - so if this won't work with flexible widgets. Is that a technical limitation, or does it not make sense conceptually?

Reply:
To scroll, you've got to have a fixed/max width, otherwise the widget won't scroll. This is why I split out `width` and `scroll_width` originally.

Reply:
> To scroll, you've got to have a fixed/max width, otherwise the widget won't scroll.

But what's the reason for this limitation?

Reply:
If the text box resizes to the width of the text then there's no point in scrolling. You scroll when the text is wider than the container.

Reply:
Huh, oh yeah :laughing: